### ISSUE TITLE
Enhancement request: update cellRenderer function signature to include itemIndex 

### ISSUE THREAD

**Issue by jmoreau-enquizit** [tabalinas/jsgrid#666]
The itemIndex is a helpful data point that is available at the row level (i.e. rowRenderer). It would also be helpful to be able to pass it in at the cell level so that we can distinguish which row a particular cell value is from when customizing the cellRenderer function.

Thanks for considering this request.

**Comment by richraid21** [comment]
What kind of use-cases do you see this being helpful for?

**Comment by jmoreau-enquizit** [comment]
I had to send the table data to the back end for processing without using AJAX and the solution I came up with required a way to specify both the cell value and cell index so that I could create a json object. 

Specifically, I created hidden input fields with the following format for each cell using cellRenderer:

<input type=hidden name='[row_index][cell_name]' value='cell_value'>

Upon submission, the parameters could easily be parsed and stored in the database and retrieved to recreate the original user input.

Thanks for following up.

**Comment by richraid21** [comment]
Ok, I could also see this coming in useful if subgrid functionality is ever added by default. 

It's not a hard feature so I doubt @tabalinas would have any objections?

**Comment by tabalinas** [comment]
Yes, I agree, this is useful functionality. No objections for this.
Thanks for your suggestion!
